ddtrace/internal/datadog/profiling/include/exporter.hpp

```diff
@@ -0,0 +1,261 @@
+  static constexpr std::string_view family = "python";
+
+  std::string errmsg;
+
+public:
+  DdogProfExporter(std::string_view env, std::string_view service,
+                   std::string_view version, std::string_view runtime,
+                   std::string_view runtime_version,
+                   std::string_view profiler_version, std::string_view url,
+                   ExporterTagset &user_tags);
+  ~DdogProfExporter();
+
+  ddog_prof_Exporter *ptr;
+};
+
+class Uploader {
+  bool agentless; // Whether or not to actually use API key/intake
+  size_t profile_seq = 0;
+  std::string runtime_id;
+  std::string url;
+
+  std::unique_ptr<DdogProfExporter> ddog_exporter;
+
+  std::string errmsg;
+
+public:
+  Uploader(std::string_view _env, std::string_view _service,
+           std::string_view _version, std::string_view _runtime,
+           std::string_view _runtime_version,
+           std::string_view _profiler_version, std::string_view _url,
+           DdogProfExporter::ExporterTagset &user_tags);
+  bool set_runtime_id(const std::string &id);
+  bool upload(const Profile *profile);
+};
+
+class UploaderBuilder {
+  std::string env;
+  std::string service;
+  std::string version;
+  std::string runtime;
+  std::string runtime_version;
+  std::string profiler_version;
+  std::string url;
+  DdogProfExporter::ExporterTagset user_tags;
+
+public:
+  UploaderBuilder &set_env(std::string_view env);
+  UploaderBuilder &set_service(std::string_view service);
+  UploaderBuilder &set_version(std::string_view version);
+  UploaderBuilder &set_runtime(std::string_view runtime);
+  UploaderBuilder &set_runtime_version(std::string_view runtime_version);
+  UploaderBuilder &set_profiler_version(std::string_view profiler_version);
+  UploaderBuilder &set_url(std::string_view url);
+  UploaderBuilder &set_tag(std::string_view key, std::string_view val);
+
+  UploaderBuilder(std::string_view _env = "prod",
+                  std::string_view _service = "py_libdatadog",
+                  std::string_view _version = "",
+                  std::string_view _runtime = "cython",
+                  std::string_view _runtime_version = "???",
+                  std::string_view _profiler_version = "???",
+                  std::string_view _url = "http://localhost:8126");
+
+  Uploader *build_ptr();
+};
+
+class Profile {
+public:
+  enum ProfileType : unsigned int {
+    CPU = 1 << 0,
+    Wall = 1 << 1,
+    Exception = 1 << 2,
+    LockAcquire = 1 << 3,
+    LockRelease = 1 << 4,
+    Allocation = 1 << 5,
+    Heap = 1 << 6,
+    All = CPU | Wall | Exception | LockAcquire | LockRelease | Allocation | Heap
+  };
+
+private:
+  unsigned int type_mask;
+  unsigned int max_nframes;
+  unsigned int nframes;
+
+  std::string errmsg;
+
+  // Keeps temporary buffer of frames in the stack
+  std::vector<ddog_prof_Location> locations;
+  std::vector<ddog_prof_Line> lines;
+
+  // Storage for strings
+  std::unordered_set<std::string> strings;
```

Comment:
Minor: having class members prefixed with `_` really makes life easier, though I agree this is a matter of taste.